src/ImageSharp/Common/Helpers/Vector128Utilities.cs

```diff
@@ -71,11 +82,11 @@ public static bool SupportsShiftByte
     /// <param name="control">The shuffle control byte.</param>
     /// <returns>The <see cref="Vector128{Single}"/>.</returns>
     [MethodImpl(MethodImplOptions.AggressiveInlining)]
-    public static Vector128<float> ShuffleNative(Vector128<float> vector, [ConstantExpected] byte control)
+    public static Vector128<int> ShuffleNative(Vector128<int> vector, [ConstantExpected] byte control)
     {
-        if (Sse.IsSupported)
+        if (Sse2.IsSupported)
         {
-            return Sse.Shuffle(vector, vector, control);
+            return Sse2.Shuffle(vector, control);
         }
```

Comment:
Similar thing here. I'd expect just the xplat call is sufficient and the JIT will continue emitting the optimal shuffle instruction for that control byte.

Reply:
I can't see `Sse2.Shuffle(vector, control)` for float?